packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.controller.js

```diff
@@ -0,0 +1,208 @@
+    this.cucRegionService = CucRegionService;
+    this.DATA_PROCESSING_STATUS_TO_CLASS = DATA_PROCESSING_STATUS_TO_CLASS;
+    this.getDataProcessingUiUrl = getDataProcessingUiUrl;
+    this.containerService = PciStoragesContainersService;
+    this.containerId = null;
+    this.metricsTimer = null;
+    this.moment = moment;
+    // setup metrics retrieval
+    this.warp10 = $resource(
+      WARP10_URL,
+      {},
+      {
+        query: {
+          method: 'POST',
+          transformRequest: [],
+          isArray: true,
+        },
+      },
+    );
+    // setup metrics container
+    this.metrics = {
+      totalMemory: {
+        data: [],
+        labels: [],
+      },
+      activeTasks: {
+        data: [],
+        labels: [],
+      },
+      blockManagerDiskUsed: {
+        data: [],
+        label: [],
+      },
+    };
+  }
+
+  $onInit() {
+    // retrieve container id for the job
+    this.containerService.getAll(this.projectId).then((containers) => {
+      const container = find(
+        containers,
+        (c) => c.name === this.job.containerName,
+      );
+      if (container) {
+        this.containerId = container.id;
+      }
+    });
+    // start metrics retrieval
+    this.queryMetrics();
+  }
+
+  $onDestroy() {
+    if (this.metricsTimer !== null) {
+      this.$timeout.cancel(this.metricsTimer);
+    }
+  }
+
+  /**
+   * Query metrics from OVH metrics backend
+   * If job is still running, we query each METRICS_REFRESH_INTERVAL to update charts
+   */
+  queryMetrics() {
+    this.queryMetricsTotalMemory();
+    this.queryMetricsActiveTasks();
+    this.queryMetricsDiskUsed();
+    if (this.job.endDate === null) {
+      this.metricsTimer = this.$timeout(
+        () => this.queryMetrics(),
+        METRICS_REFRESH_INTERVAL,
+      );
+    }
+  }
+
+  /**
+   * Compute start date and end dates for charts depending on job status.
+   * If job is running, we plot the last 5 minutes.
+   * If job is not running any more, we plot the last 5 minutes before the end of job
+   * @return {{endDate: moment.Moment, startDate: moment.Moment}}
+   */
+  computeDates() {
+    let startDate;
+    let endDate;
+    if (this.isJobRunning()) {
+      startDate = moment().subtract(5, 'minutes');
+      endDate = moment();
+    } else {
+      startDate = moment(this.job.endDate).subtract(5, 'minutes');
+      endDate = moment(this.job.endDate);
+    }
+    return {
+      startDate,
+      endDate,
+    };
+  }
+
+  /**
+   * Retrieve JVM total memory usage for current job from metrics backend
+   */
+  queryMetricsTotalMemory() {
+    const d = this.computeDates();
+    this.warp10
+      .query(
+        `[ '${
+          this.metricsToken.data.token
+        }' 'spark_jvm_memory_usage' { 'qty' 'used' 'mem_type' 'total' 'job-id' '${
+          this.job.id
+        }' } '${d.startDate.toISOString()}' '${d.endDate.toISOString()}' ] FETCH SORT [ SWAP [ 'executor-id' ] reducer.sum ] REDUCE`,
+      )
```

Comment:
I'm having trouble understanding the query here, probably because it is a special format. Maybe this could a function to format the `query` so the format could be abstracted